course/api/full_serializers.py

```diff
@@ -36,29 +34,35 @@
 ]
 
 
-class CourseModuleSerializer(AplusModelSerializer):
-    url = NestedHyperlinkedIdentityField(view_name='api:course-exercises-detail')
-    exercises = serializers.SerializerMethodField()
-    display_name = serializers.CharField(source='__str__')
 
-    class Meta(AplusModelSerializer.Meta):
-        model = CourseModule
-        fields = (
-            'url',
-            'html_url',
-            'display_name',
-            'is_open',
-            'exercises',
-        )
+class ModuleExerciseSerializer(serializers.Serializer):
+    id =  serializers.IntegerField()
+    url = serializers.CharField()
+    html_url = serializers.CharField()
+    display_name = serializers.CharField()
+    max_points = serializers.IntegerField()
+    max_submissions = serializers.IntegerField()
+    hierarchical_name = serializers.CharField()
+    difficulty = serializers.CharField()
 
+class CourseModuleSerializer(serializers.Serializer):
+    """
+    Serializes items in the `modules` list of `CachedContent.data`. Does not
+    derive from `AplusModelSerializer` because the items are `dict`s instead of
+    model objects.
+    """
+    id = serializers.IntegerField()
+    url = serializers.CharField()
+    html_url = serializers.CharField()
+    display_name = serializers.CharField()
+    is_open = serializers.BooleanField()
+    exercises = serializers.SerializerMethodField()
     def get_exercises(self, obj: CourseModule) -> OrderedDict:
-        # List only exercises derived from BaseExercise, thus no Chapter texts (/exercise/<id> returns 404 for those)
-        # Check if the exercises were prefetched in the view
-        if hasattr(obj, 'exercises'):
-            exercises = obj.exercises
-        else:
-            exercises = BaseExercise.objects.filter(course_module=obj).all()
-        serializer = ExerciseBriefSerializer(instance=exercises, many=True, context=self.context)
+        serializer = ModuleExerciseSerializer(
```

Comment:
At first sight, it looks like this file doesn't change any functionality. The old DRY code was replaced with duplicated field definitions that do basically the same thing as the old code. What was the purpose of this?